source/library/Patrol/Client.hs

```diff
@@ -3,12 +3,52 @@ module Patrol.Client where
 import qualified Control.Monad.Catch as Catch
 import qualified Control.Monad.IO.Class as IO
 import qualified Data.Aeson as Aeson
+import qualified Data.Text as Text
+import qualified GHC.Stack as Stack
 import qualified Network.HTTP.Client as Client
+import qualified Network.HTTP.Client.TLS as Tls
 import qualified Patrol.Exception.Problem as Problem
 import qualified Patrol.Type.Dsn as Dsn
+import qualified Patrol.Type.Envelope as Envelope
 import qualified Patrol.Type.Event as Event
 import qualified Patrol.Type.Response as Response
+import qualified System.Environment as Environment
 
+-- | Capture an exception by sending it to Sentry. The DSN is read from the
+-- @SENTRY_DSN@ environment variable. To customize the behavior, use
+-- 'captureExceptionWith'.
+captureException ::
+  (Catch.Exception e, IO.MonadIO io, Catch.MonadThrow io) =>
+  e ->
+  io Response.Response
+captureException e = do
+  dsn <- do
+    maybeString <- IO.liftIO $ Environment.lookupEnv "SENTRY_DSN"
+    Dsn.fromText $ maybe Text.empty Text.pack maybeString
+  captureExceptionWith (const Nothing) pure dsn e
+
+captureExceptionWith ::
+  (Catch.Exception e, IO.MonadIO io, Catch.MonadThrow io) =>
+  -- | How to get a 'Stack.CallStack' from a 'Catch.SomeException'. Use
+  -- @'const' 'Nothing'@ if you don't want to get a call stack.
+  (Catch.SomeException -> Maybe Stack.CallStack) ->
+  -- | How to modify the 'Envelope.Envelope' before it is sent. Use @'pure'@ if
+  -- you don't want to modify the envelope.
+  (Envelope.Envelope -> io Envelope.Envelope) ->
```

Comment:
My intent was to allow modifying the envelope just before sending. However this argument is less useful than expected because the envelope is already encoded. So if you want to change something about the event you're out of luck. I think this argument should be removed. If someone wants to customize the envelope and/or request, they should use the primitives already provided. That way this function is just a version of `captureException` that allows you to customize the call stack and DSN.